### PR TITLE
[Feature] Buttons update

### PR DESCRIPTION
**Pull request by seagyn**
Some time was spent updating some of the dependencies which had a couple of vulnerabilities. There are still a handful but they have been patched upstream and waiting for them to come down.

Updated the buttons into one component and updated the storybook to show the different options.

### CODE REVIEW

File: src/button/index.js [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
-export {
-  ButtonPrimary,
-  ButtonSecondary,
-  ButtonWhite,
-  ButtonPill,
-  ButtonWarning,
-  ButtonOcean,
-  ButtonDark,
-  LoadingIcon
+const Button = props => {
```
Comment (MrOpperman) [inline src/button/index.js]: I would refactor this, basically removes the need for the return function. Are all the props required to be passed to StyledButton? Like children and loading?

```
const Button = ({ children, loading, ...rest }) => (
   <StyledButton {...rest} children={children} loading={loading}>
    ....
);
```
or 
```
const Button = ({ children, loading, ...props }) => (
   <StyledButton {...props}>
    ....
);
```

File: src/button/index.js [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
+    <StyledButton {...props}>
+      {children}
+      {loading && <Loading />}
+    </StyledButton>
+  );
 };
+
+Button.propTypes = {
+  color: PropTypes.oneOf(['primary', 'secondary', 'warning', 'success']),
+  children: PropTypes.any.isRequired,
```
Comment (MrOpperman) [inline src/button/index.js]: Is this basically text/svg?

Comment (seagyn) [reply src/button/index.js]: Yes, that's correct. I wasn't entirely sure how to support both. The other option I thought of was passing down an icon prop and then children would always be text.

Comment (MrOpperman) [reply src/button/index.js]: I think this is fine, icon in the props is fine too.

Comment (seagyn) [reply src/button/index.js]: So this is proving a little harder than I expected with the SVG loader and passing it down. I'll move this to a later feature.

**Comment by seagyn** [comment]
@MrOpperman I pushed a couple commits. Nothing major. Could you review and let me know if I can merge.

**Pull request merged**